test/shared/src/test/scala/zio/test/TestMain.scala

```diff
@@ -4,29 +4,44 @@ import scala.concurrent.ExecutionContext.Implicits.global
 
 import zio.test.mock._
 import zio.test.TestUtils.{ report, scope }
+import scala.collection.immutable.ListMap
 
 object TestMain {
 
   def main(args: Array[String]): Unit = {
-    val testResults = List(
-      scope(AssertionSpec.run, "AssertionSpec"),
-      scope(BoolAlgebraSpec.run, "BoolAlgebraSpec"),
-      scope(CheckSpec.run, "CheckSpec"),
-      scope(ClockSpec.run, "ClockSpec"),
-      scope(ConsoleSpec.run, "ConsoleSpec"),
-      scope(DefaultTestReporterSpec.run, "DefaultTestReporterSpec"),
-      scope(EnvironmentSpec.run, "EnvironmentSpec"),
-      scope(FunSpec.run, "FunSpec"),
-      scope(GenSpec.run, "GenSpec"),
-      scope(GenZIOSpec.run, "GenZIOSpec"),
-      scope(LiveSpec.run, "LiveSpec"),
-      scope(RandomSpec.run, "RandomSpec"),
-      scope(SampleSpec.run, "SampleSpec"),
-      scope(SchedulerSpec.run, "SchedulerSpec"),
-      scope(SystemSpec.run, "SystemSpec"),
-      scope(TestAspectSpec.run, "TestAspectSpec"),
-      scope(TestSpec.run, "TestSpec")
+    val allTests: ListMap[String, BaseSpec] = ListMap(
```

Comment:
Can we just use a `List` here and do `List.filter(_._2 == selection)`?

Reply:
Sure, just though this list has the semantic of a map, so maybe I should use one. But `ListMap` isn't that commonly known and used and it doesn't provide any big benefit in the code. I'll revert it back to a list.